packages/@dcl/ecs/src/systems/raycast.ts

```diff
@@ -0,0 +1,244 @@
+  /**
+   * @public
+   * Execute callback when the entity receives a RaycastResult component update.
+   * Uses a Vector3 global direction value to calculate the ray direction
+   * @param entity - Entity to attach the callback
+   * @param callback - Function to execute when the entity's RaycastResult component is updated
+   * @param options - Raycast configuration options
+   */
+  registerGlobalDirectionRaycast(
+    entity: Entity,
+    callback: RaycastSystemCallback,
+    options?: Partial<GlobalDirectionRaycastOptions>
+  ): void
+
+  /**
+   * @public
+   * Execute callback when the entity receives a RaycastResult component update.
+   * Uses a Vector3 global target position to calculate the ray direction
+   * @param entity - Entity to attach the callback
+   * @param callback - Function to execute when the entity's RaycastResult component is updated
+   * @param options - Raycast configuration options
+   */
+  registerGlobalTargetRaycast(
+    entity: Entity,
+    callback: RaycastSystemCallback,
+    options?: Partial<GlobalTargetRaycastOptions>
+  ): void
+
+  /**
+   * @public
+   * Execute callback when the entity receives a RaycastResult component update.
+   * Uses an target Entity value to calculate the ray direction
+   * @param entity - Entity to attach the callback
+   * @param callback - Function to execute when the entity's RaycastResult component is updated
+   * @param options - Raycast configuration options
+   */
+  registerTargetEntityRaycast(
+    entity: Entity,
+    callback: RaycastSystemCallback,
+    options?: Partial<TargetEntityRaycastOptions>
+  ): void
+}
+
+/**
+ * @internal
+ */
+export function createRaycastSystem(engine: IEngine): RaycastSystem {
+  const raycastComponent = components.Raycast(engine)
+  const raycastResultComponent = components.RaycastResult(engine)
+  const entitiesCallbackResultMap = new Map<Entity, { callback: RaycastSystemCallback }>()
+
+  const defaultOptions: RaycastSystemOptions = {
+    maxDistance: 16,
+    queryType: RaycastQueryType.RQT_HIT_FIRST,
+    continuous: false,
+    originOffset: { x: 0, y: 0, z: 0 },
+    collisionMask: ColliderLayer.CL_PHYSICS
+  }
+  const getLocalDirectionRaycastDefaultOptions = (
+    options: Partial<LocalDirectionRaycastOptions> = {}
+  ): RaycastSystemOptions => ({
+    ...defaultOptions,
+    directionRawValue: {
+      $case: 'localDirection',
+      localDirection: options.direction || { x: 0, y: 0, z: 1 }
+    },
+    ...options
+  })
+  const getGlobalDirectionRaycastDefaultOptions = (
+    options: Partial<GlobalDirectionRaycastOptions> = {}
+  ): RaycastSystemOptions => ({
+    ...defaultOptions,
+    directionRawValue: {
+      $case: 'globalDirection',
+      globalDirection: options.direction || { x: 0, y: 0, z: 1 }
+    },
+    ...options
+  })
+  const getGlobalTargetRaycastDefaultOptions = (
+    options: Partial<GlobalTargetRaycastOptions> = {}
+  ): RaycastSystemOptions => ({
+    ...defaultOptions,
+    directionRawValue: {
```

Comment:
if we dont want that directionRawValue can be changed, then it should be declared after the `...options`. If not i can override it

Reply:
since the `direcitonRawValue` is internal it's not exposed when the options parameter is set in the scene code, but just in case I'll make the change you suggest, nice catch!